contracts/InterchainTokenService.sol

```diff
@@ -796,12 +810,21 @@ contract InterchainTokenService is
      * @param gasValue The amount of gas to be paid for the transaction.
      */
     function _callContract(
-        string calldata destinationChain,
+        string memory destinationChain,
         bytes memory payload,
         MetadataVersion metadataVersion,
         uint256 gasValue
     ) internal {
         string memory destinationAddress = trustedAddress(destinationChain);
+
+        // Check if the ITS call should be routed via ITS hub for this destination chain
+        if (keccak256(abi.encodePacked(destinationAddress)) == ITS_HUB_TRUSTED_ADDRESS_HASH) {
```

Comment:
Nice enough that this can also be moved to a different contract to not cause additional code bloat.